src/main/java/com/synopsys/integration/detect/lifecycle/run/step/ContainerScanStepRunner.java

```diff
@@ -4,25 +4,95 @@
 import java.io.IOException;
 import java.util.UUID;
 
-import com.synopsys.integration.detect.configuration.DetectUserFriendlyException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.synopsys.integration.detect.lifecycle.run.data.BlackDuckRunData;
 import com.synopsys.integration.detect.lifecycle.run.operation.OperationRunner;
-import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioUtil;
+import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioHeaderUtil;
 import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.response.Response;
+import com.synopsys.integration.util.NameVersion;
+
+import com.google.gson.JsonObject;
 
 public class ContainerScanStepRunner {
 
     private final OperationRunner operationRunner;
     private UUID scanId;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final NameVersion projectNameVersion;
+    private final String projectGroupName;
+    private final BlackDuckRunData blackDuckRunData;
+    private final File binaryRunDirectory;
 
-    public ContainerScanStepRunner(OperationRunner operationRunner) throws DetectUserFriendlyException {
+    public ContainerScanStepRunner(OperationRunner operationRunner, NameVersion projectNameVersion, BlackDuckRunData blackDuckRunData) {
         this.operationRunner = operationRunner;
+        this.projectNameVersion = projectNameVersion;
+        this.blackDuckRunData = blackDuckRunData;
+        binaryRunDirectory = operationRunner.getDirectoryManager().getBinaryOutputDirectory();
+        projectGroupName = operationRunner.calculateProjectGroupOptions().getProjectGroup();
     }
 
-    public UUID initiateScan(BlackDuckRunData blackDuckRunData) throws IOException, IntegrationException {
-        DetectProtobufBdioUtil detectProtobufBdioUtil = new DetectProtobufBdioUtil(UUID.randomUUID().toString(), "CONTAINER");
-        File bdioHeaderFile = detectProtobufBdioUtil.createProtobufBdioHeader();
-        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
+    public UUID invokeContainerScanningWorkflow() throws IntegrationException, IOException {
+        initiateScan();
+        uploadImageToStorageService();
+        uploadImageMetadataToStorageService();
         return scanId;
     }
+
+    public void initiateScan() throws IOException, IntegrationException {
+        DetectProtobufBdioHeaderUtil detectProtobufBdioHeaderUtil = new DetectProtobufBdioHeaderUtil(UUID.randomUUID().toString(), "CONTAINER", projectNameVersion, projectGroupName);
+        File bdioHeaderFile = detectProtobufBdioHeaderUtil.createProtobufBdioHeader(binaryRunDirectory);
+        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
```

Comment:
Is scanId validated for valid UUID?

Reply:
Yes. ScanID is extracted from the response using the `UUID.fromString` method which will throw an exception if it's an invalid UUID.